### PR TITLE
Add register_*_effect to allow registering custom effects

### PR DESCRIPTION
**Pull request by ayufan**
## Description:

This allows to register custom effect from user components,
allowing for bigger composability of source.

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by ayufan** [comment]
This is needed by https://github.com/esphome/esphome/pull/948

**Comment by ayufan** [comment]
@OttoWinter WDYT?

File: esphome/components/light/effects.py [hunk 1]
```diff
@@ -245,6 +271,7 @@ def validator(value):
         for i, x in enumerate(value):
             key = next(it for it in x.keys())
             if key not in allowed_effects:
+                print("KEY: {}: {}".format(key, allowed_effects))
```
Comment (OttoWinter) [inline esphome/components/light/effects.py]: Stale debug statement

**Review by OttoWinter — APPROVED** [review]
LGTM, thanks!

One thing: the print debug statement should be removed

**Comment by ayufan** [comment]
@OttoWinter done

File: esphome/components/light/effects.py (hunk 1 above)
Comment (OttoWinter) [reply esphome/components/light/effects.py]: The debug statement is still there :)

Comment (ayufan) [reply esphome/components/light/effects.py]: Sorry :)

**Review by ayufan — COMMENTED** [review]
Fixed, and sorry.

**Review by OttoWinter — COMMENTED** [review]
Ups, looks like there's unfortunately still a bit left: the linter expects two blank lines after a function definition, but here there's only one: https://travis-ci.org/esphome/esphome/jobs/635988641?utm_medium=notification&utm_source=github_status

But when lint passes then this is ok to merge :+1:

**Comment by ayufan** [comment]
> Ups, looks like there's unfortunately still a bit left: the linter expects two blank lines after a function definition, but here there's only one: https://travis-ci.org/esphome/esphome/jobs/635988641?utm_medium=notification&utm_source=github_status
> 
> But when lint passes then this is ok to merge

OK. I don't see it locally. Let me fix it.

**Pull request merged**